fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordKeyCursorTest.java

```diff
@@ -0,0 +1,270 @@
+            // index cannot be used with large fields
+            metaData.removeIndex("MySimpleRecord$str_value_indexed");
+        };
+    }
+
+    @Nullable
+    private static ScanProperties getScanProperties(final UseContinuations useContinuations) {
+        ExecuteProperties executeProperties;
+        switch (useContinuations) {
+            case NONE:
+                return null;
+            case CONTINUATIONS:
+                executeProperties = ExecuteProperties.newBuilder()
+                        .setReturnedRowLimit(19)
+                        .setIsolationLevel(IsolationLevel.SERIALIZABLE)
+                        .build();
+                return new ScanProperties(executeProperties);
+            case BYTE_LIMIT:
+                executeProperties = ExecuteProperties.newBuilder()
+                        .setScannedBytesLimit(2000)
+                        .setIsolationLevel(IsolationLevel.SERIALIZABLE)
+                        .build();
+                return new ScanProperties(executeProperties);
+            default:
+                throw new IllegalArgumentException("Unknown continuation");
+        }
+    }
+
+    private List<Tuple> scanKeys(@Nonnull FDBRecordStore store, @Nullable ScanProperties scanProperties, boolean withContinuations) throws InterruptedException, ExecutionException {
+        if (scanProperties == null) {
+            scanProperties = ScanProperties.FORWARD_SCAN;
+        }
+        RecordCursor<Tuple> recordKeyCursor = store.scanRecordKeys(TupleRange.allOf(null), null, scanProperties);
+        if (!withContinuations) {
+            return recordKeyCursor.asList().get();
+        } else {
+            boolean done = false;
+            List<Tuple> result = new ArrayList<>();
+            while (!done) {
+                RecordCursorResult<Tuple> currentRecord = recordKeyCursor.getNext();
+                while (currentRecord.hasNext()) {
+                    result.add(currentRecord.get());
+                    currentRecord = recordKeyCursor.getNext();
+                }
+                RecordCursorContinuation continuation = currentRecord.getContinuation();
+                if (continuation.isEnd()) {
+                    done = true;
+                } else {
+                    recordKeyCursor = store.scanRecordKeys(TupleRange.allOf(null), continuation.toBytes(), scanProperties);
+                }
+            }
+            return result;
+        }
+    }
+
+    @Nonnull
+    private List<FDBStoredRecord<Message>> saveRecords(final boolean splitLongRecords, final int formatVersion, final RecordMetaDataHook hook) throws Exception {
+        List<FDBStoredRecord<Message>> result;
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            List<FDBStoredRecord<Message>> result1 = new ArrayList<>(100);
+            for (int i = 1; i <= 100; i++) {
```

Comment:
You have a bunch of code that does `IntStream.range(1, 101)`, it would probably be worthwhile to have a helper method to get the list of `recNo` to be saved (as a stream), and then use that here, and everywhere else.